include/seqan3/alignment/pairwise/execution/execution_handler_parallel.hpp

```diff
@@ -99,15 +99,14 @@ class execution_handler_parallel
 
     /*!\brief Takes underlying range of sequence pairs and invokes an alignment on each instance.
      * \tparam algorithm_t              The type of the alignment algorithm.
-     * \tparam indexed_sequence_pairs_t The type of underlying sequence pairs annotated with an index;
-     *                                  must model seqan3::detail::indexed_sequence_pair_range.
+     * \tparam indexed_sequence_pairs_t The type of underlying sequence pairs annotated with an index.
      * \tparam delegate_type            The type of the callable invoked on the std::invoke_result of `algorithm_t`.
      *
      * \param[in] algorithm              The alignment algorithm to invoke.
      * \param[in] indexed_sequence_pairs The range of underlying annotated sequence pairs to be aligned.
      * \param[in] delegate               A callable which will be invoked on each result of the computed alignments.
      */
-    template <typename algorithm_t, indexed_sequence_pair_range indexed_sequence_pairs_t, typename delegate_type>
+    template <typename algorithm_t, typename indexed_sequence_pairs_t, typename delegate_type>
```

Comment:
Shouldn't the name change? Like `data_t`, `work_data_t`, `algorithm_data_t`?

Is the assumption that you move in the data still valid?

Reply:
Honestly, all these questions can be answered when we know the sructure and the data of the search algorithm. Also then it is the question if we can define a common requirement on the type. Or if this has to be a variadic template etc.

Reply:
But for now it still exists within the context of alignments. But I wanted to separate the structural changes from the documentation/location changes.